moshi/src/main/resources/META-INF/proguard/moshi.pro

```diff
@@ -16,6 +16,16 @@
 # The name of @JsonClass types is used to look up the generated adapter.
 -keepnames @com.squareup.moshi.JsonClass class *
 
+# Retain generated target class's synthetic defaults constructor and keep DefaultConstructorMarker's
+# name. We will look this up reflectively to invoke the type's constructor.
+#
+# We can't _just_ keep the defaults constructor because Proguard/R8's spec doesn't allow wildcard
+# matching preceding parameters.
+-keepnames class kotlin.jvm.internal.DefaultConstructorMarker
+-keepclassmembers @com.squareup.moshi.JsonClass class * {
```

Comment:
Should we match on the metadata annotation as well to prevent affecting Java classes?

Reply:
Good call, since this is suggested for keeping enum members too. Will fix

Reply:
Part of #907